src/apt_cache.rs

```diff
@@ -68,9 +68,13 @@ pub fn policies(lines: impl Stream<Item = io::Result<String>>) -> impl Stream<It
                             .or_insert_with(Vec::new)
                             .push(source.trim().to_owned());
                     } else if let Some(version) = line.strip_prefix(" *** ") {
-                        current_version = version.trim().to_owned();
+                        let mut current_version_and_pin = version.trim().split_whitespace();
+                        current_version = current_version_and_pin.next().unwrap().to_owned();
+                        // TODO: Store & use pin.
                     } else if let Some(version) = line.strip_prefix("   ") {
-                        current_version = version.trim().to_owned();
+                        let mut current_version_and_pin = version.trim().split_whitespace();
+                        current_version = current_version_and_pin.next().unwrap().to_owned();
```

Comment:
Since unwrapping creates a panic on `None`, it may be better to ignore lines that for some reason have no version.

```rs
if let Some(version) =  current_version_and_pin.next() {
    current_version = version.to_owned();
}
```

Reply:
Seems more important for the second invocation of `.next()` since getting into the `if let Some(version) = line.strip_prefix("   ")` block means there's at least one string we're working with, but I guess this would guard against if a line was _just_ the prefix we're stripping/other whitespace?

Reply:
Yes, this would guard against the line having a prefix but nothing after. Alternatively, we could set an `.expect("")` here explain why the unwrap should never fail.